src/lib/autodiscovery/autodiscovery.py

```diff
@@ -0,0 +1,110 @@
+import os
+import time
+from dataclasses import asdict
+from typing import List
+
+from aiohttp import ClientSession
+
+from lib.autodiscovery.gcp_metrics_descriptor import GCPMetricDescriptor
+from lib.clientsession_provider import init_gcp_client_session
+from lib.configuration import config
+from lib.context import LoggingContext
+from lib.credentials import create_token
+from lib.metrics import GCPService, Metric
+
+logging_context = LoggingContext("AUTODISCOVERY")
+
+
+discovered_resource_type = os.environ.get("AUTODISCOVERY_RESOURCE_TYPE", "gce_instance")
+
+
+async def get_metric_descriptors(
+    gcp_session: ClientSession, token: str
+) -> List[GCPMetricDescriptor]:
+    project_id = config.project_id()
+    headers = {"Accept": "application/json", "Authorization": f"Bearer {token}"}
+    url = f"https://monitoring.googleapis.com/v3/projects/{project_id}/metricDescriptors"
+
+    response = await gcp_session.request("GET", url=url, headers=headers)
+    response = await response.json()
+
+    discovered_metrics_descriptors = [
+        GCPMetricDescriptor(**descriptor) for descriptor in response["metricDescriptors"]
```

Comment:
Are you sure that first (and only) request will fetch all? What if there are many metrics in env? Is there paging there? But you are not fetching it?